src/helics/shared_api_library/Translators.h

```diff
@@ -0,0 +1,298 @@
+
+/**
+ * Get a translator by its name, typically already created via registerInterfaces file or something of that nature.
+ *
+ * @param fed The federate object to use to get the translator.
+ * @param name The name of the translator.
+ *
+ * @param[in,out] err The error object to complete if there is an error.
+
+ *
+ * @return A HelicsTranslator object, the object will not be valid and err will contain an error code if no translator with the specified
+ name
+ * exists.
+ */
+HELICS_EXPORT HelicsTranslator helicsFederateGetTranslator(HelicsFederate fed, const char* name, HelicsError* err);
+/**
+ * Get a translator by its index, typically already created via registerInterfaces file or something of that nature.
+ *
+ * @param fed The federate object in which to create a publication.
+ * @param index The index of the publication to get.
+ *
+ * @param[in,out] err A pointer to an error object for catching errors.
+
+ *
+ * @return A HelicsTranslator, which will be NULL if an invalid index is given.
+ */
+HELICS_EXPORT HelicsTranslator helicsFederateGetTranslatorByIndex(HelicsFederate fed, int index, HelicsError* err);
+
+/**
+ * Check if a translator is valid.
+ *
+ * @param trans The translator object to check.
+ *
+ * @return HELICS_TRUE if the Translator object represents a valid translator.
+ */
+HELICS_EXPORT HelicsBool helicsTranslatorIsValid(HelicsTranslator trans);
+
+/**
+ * Get the name of the translator and store in the given string.
+ *
+ * @param trans The given translator.
+ *
+ * @return A string with the name of the translator.
+ */
+HELICS_EXPORT const char* helicsTranslatorGetName(HelicsTranslator trans);
+
+/**
+ * Set a property on a translator.
+ *
+ * @param trans The translator to modify.
+ * @param prop A string containing the property to set.
+ * @param val A numerical value for the property.
+ *
+ * @param[in,out] err A pointer to an error object for catching errors.
+
+ */
+HELICS_EXPORT void helicsTranslatorSet(HelicsTranslator trans, const char* prop, double val, HelicsError* err);
+
+/**
+ * Set a string property on a translator.
+ *
+ * @param trans The translator to modify.
+ * @param prop A string containing the property to set.
+ * @param val A string containing the new value.
+ *
+ * @param[in,out] err A pointer to an error object for catching errors.
+
+ */
+HELICS_EXPORT void helicsTranslatorSetString(HelicsTranslator trans, const char* prop, const char* val, HelicsError* err);
+
+/**
+ * Add an input to send a translator output.
+ *
+ * @details All messages going to a destination are copied to the delivery address(es).
+ * @param trans The given translator to add a destination target to.
+ * @param dst The name of the endpoint to add as a destination target.
+ *
+ * @param[in,out] err A pointer to an error object for catching errors.
+
+ */
+HELICS_EXPORT void helicsTranslatorAddInputTarget(HelicsTranslator trans, const char* input, HelicsError* err);
```

Comment:
I'm confused by the language used in the API name vs how it is described in the comments. I feel like I understand what the API names are doing until I read the documentation comments.